### ISSUE TITLE
libssl issue

### ISSUE THREAD

**Issue by nkrabben** [claunia/DiscImageChef#757]
### Version

5.3.0

### Tested debug version?

- [X] Yes

### Which operating systems have you used?

- [ ] Windows
- [X] Linux
- [ ] macOS
- [ ] Other

### What is the architectural bit size you're using?

- [ ] 32-bit
- [ ] 64-bit
- [X] Unsure or unknown

### What processor are you using?

- [X] An Intel or AMD
- [ ] An ARM or Apple Silicon
- [ ] Unsure or unknown

### Description

We have a fresh install of Ubuntu 22 LTS. I downloaded the [aaru-5.3.1_linux_amd64.tar.gz](https://github.com/aaru-dps/Aaru/releases/download/v5.3.1/aaru-5.3.1_linux_amd64.tar.gz) release, unpacked the files, opened a terminal in the folder and ran `./aaru`. 

I receive the same error as documented in #238 

I've tried to install the dotnet6 runtime and sdk, but neither fix the issue.

I'm assuming that I'm missing a dependency or I need to do something else to install Aaru, but I'm not sure what.

### Exact command line used

./aaru

### Expected behavior

Help text or usage instructions

### Actual behavior

```
No usable version of the libssl was found
Aborted (core dumped)
```

### Output of command execution with debug output enabled

```shell
Same as above
```

**Comment by robin-francois** [comment]
Hello, have you tried the workaround in #238:
https://github.com/aaru-dps/Aaru/issues/238#issuecomment-540225801

**Comment by claunia** [comment]
This is unfortunately a cat and mouse situation.

When a distribution updates their earliest available library of a bunch of dotnet necessary components (ICU and OpenSSL amongst others), we must wait for the dotnet update to enhance their distribution support (if done in the current release and not on the next one only).

Then we must wait for our package creation library to update accordingly, and finally we must re-build and re-package.

So until 5.3.2 I'm afraid we cannot really help you beyond a, not very useful: compile by yourself.

Or the workaround @robin-francois commented.

In any case we plan to release 5.3.2 before 24th December.

Sorry for the inconveniences.

**Comment by nkrabben** [comment]
No worries. Makes sense.

This is shifting off-topic, but in terms of compiling myself, do I do that by downloading the source and running `build.sh`? I tried that, and I ran into a bunch of warnings about empty folders because of the subrepositories. Not sure if those should be present in the release package.

I'll try to run a recursive clone today and see if I can build successfully.

**Comment by claunia** [comment]
`build.sh` is to build the release packages.

You need to clone the repo with `git clone --recursive` and then do `dotnet build Aaru.sln`. That will compile a version only usable on that computer.

**Comment by nkrabben** [comment]
For anyone else figuring out how to compile.
```
git clone --recursive <remote>
cd Aaru
dotnet build Aaru.sln
Aaru/bin/Debug/dotnet6/aaru
```

I'm assuming paths may differ depending on the environment.

**Comment by claunia** [comment]
https://github.com/quamotion/dotnet-packaging/issues/222 seems there's a fix, I'll apply it for 5.3.2

**Comment by claunia** [comment]
Sorry even with that workaround, Aaru 5 is built with .NET Core 3.1 and that can only work with OpenSSL 1.1.

It works if you manually install the package from an older release of Ubuntu but this is not a supported scenario.

Aaru 6 will support Ubuntu 22.
